tests/samplers_tests/test_cmaes.py

```diff
@@ -280,3 +358,47 @@ def test_call_after_trial_of_base_sampler() -> None:
     ) as mock_object:
         study.optimize(lambda _: 1.0, n_trials=1)
         assert mock_object.call_count == 1
+
+
+def test_is_compatible_search_space() -> None:
+    transform = _SearchSpaceTransform(
+        {
+            "x0": optuna.distributions.UniformDistribution(2, 3),
+            "x1": optuna.distributions.CategoricalDistribution(["foo", "bar", "baz", "qux"]),
+        }
+    )
+
+    assert optuna.samplers._cmaes._is_compatible_search_space(
+        transform,
+        {
+            "x1": optuna.distributions.CategoricalDistribution(["foo", "bar", "baz", "qux"]),
+            "x0": optuna.distributions.UniformDistribution(2, 3),
+        },
+    )
+
+    # same search space size, but different param names
+    assert not optuna.samplers._cmaes._is_compatible_search_space(
+        transform,
+        {
+            "x0": optuna.distributions.UniformDistribution(2, 3),
+            "foo": optuna.distributions.DiscreteUniformDistribution(2, 3, q=0.1),  #
```

Comment:
If we follow the code comment, the distribution of "foo" should be `CategoricalDistribution` of `["foo", "bar", "baz", "qux"]`. 
```suggestion
            "foo": optuna.distributions.CategoricalDistribution(["foo", "bar", "baz", "qux"]),
```